kclvm/tools/src/LSP/src/rename.rs

```diff
@@ -1,16 +1,115 @@
 use crate::{
     from_lsp::kcl_pos,
     goto_def::find_def_with_gs,
-    util::{build_word_index_for_file_paths, parse_param_and_compile, Param},
+    util::{build_word_index_for_file_paths, parse_param_and_compile, read_file, Param},
 };
 use anyhow::{anyhow, Result};
 use kclvm_ast::ast;
 use kclvm_error::diagnostic;
 use kclvm_query::selector::parse_symbol_selector_spec;
-use lsp_types::{Location, TextEdit, Url};
-use std::collections::HashMap;
+use lsp_types::{Location, Position, Range, TextEdit, Url};
+use std::collections::{HashMap, HashSet};
+use std::fs;
 use std::path::PathBuf;
 
+pub fn rename_symbol_on_file(
```

Comment:
Could you pls add more documents and CI failed? cc @amyXia1994

Reply:
fixed